### PR TITLE
Annotations: fix page numbers

### PR DESCRIPTION
**Pull request by hius07**
(1) Fix updating page numbers when partial rendering enabled.
(2) Add support of pdf custom hidden flows.
(3) Add updating page numbers when custom hidden flows changed.

- - -
This change is [Reviewable](https://reviewable.io/reviews/koreader/koreader/12324)

### CODE REVIEW

**Review by Frenzie — APPROVED** [review]

**Comment by ichnilatis-gr** [comment]
> (2) Add support of pdf custom hidden flows. closes #12320.

Does this support the case where there are no hidden flows? I mentioned hidden flows as an idea for how some pages of the book could be re-numbered, i.e. marking some pages and giving to them a new number range.

**Comment by hius07** [comment]
> Does this support the case where there are no hidden flows? I mentioned hidden flows as an idea for how some pages of the book could be re-numbered, i.e. marking some pages and giving to them a new number range.

This PR provides annotations/exporting support of the existing features (custom hidden flows).

Can you re-number pages without creating hidden flows?
Re-number - I mean to see custom page numbers in the status bar.
Page numbers in the Bookmark list just copy them from the status bar.

Maybe I got your FR wrong (noticed highlights only) and we should keep it open.

**Comment by ichnilatis-gr** [comment]
Yes, I didn't mean just to make the page numbering in the Bookmark list match the page numbering in the status bar after creating hidden flows. I was asking for a way to re-number the pages of a book, so that the page numbering in the status bar corresponds to the actual page numbering and, consequently, the page numbering in the Bookmark list (and thus in the highlight export) corresponds to the new page numbering.

**Comment by poire-z** [comment]
> I was asking for a way to re-number the pages of a book, so that the page numbering in the status bar corresponds to the actual page numbering 

Probably what the links in my https://github.com/koreader/koreader/issues/10848#issuecomment-1694618286 were about ? (not yet implemented).

File: frontend/apps/reader/modules/readerhandmade.lua [hunk 1]
```diff
@@ -552,6 +555,7 @@ function ReaderHandMade:isInHiddenFlow(pageno)
 end
 
 function ReaderHandMade:toggleHiddenFlow(pageno)
+    self.ui.annotation:setNeedsUpdateFlag()
     local idx, is_match = self:_getItemIndex(self.flow_points, pageno)
```
Comment (poire-z) [inline frontend/apps/reader/modules/readerhandmade.lua:559]: Any idea why this one was needed here - while a `self.view.footer:onUpdateFooter` is not ? In your other 3 additions of this, they seem to go together.
(Not really familiar any longer with ReaderHandmade, and not really able to get a better overview of it right now - but it feels adding external flagging with self.ui.annotations... in this quite internal toggleHiddenFlow may be odd.)

**Review by poire-z — APPROVED** [review]
Trusting you.

File: frontend/apps/reader/modules/readerhandmade.lua (hunk 1 above)
Comment (hius07) [reply frontend/apps/reader/modules/readerhandmade.lua:559]: `toggleHiddenFlow()` is called in the Page browser only (long-press on a page thumbnail).
The footer is updated on closing the Page browser.

**Pull request merged**